File: ts_sdk/src/resource_management/fs_records/claude/claude-settings-json/attribution.ts
/**
 * ClaudeAttributionFsRecord — attribution block from settings.json.
 */
import { FsRecord, type FsRecordData } from '../../fs-record';
import { RecordType } from '../../record-types';
import { StorageLayout } from '../../storage-layout';
import { fsRecordTypeRegistry } from '../../record-type-registry';

export class ClaudeAttributionFsRecord extends FsRecord {
  static override _recordType = RecordType.CLAUDE_SETTINGS_JSON_ATTRIBUTION;
  static override _readOnly = false;
  static override _storageLayout = StorageLayout.FILE;

  commit = '';
  pr = '';

  constructor(data?: Partial<FsRecordData>) {
    super(data);
  }
}

fsRecordTypeRegistry.register(
  ClaudeAttributionFsRecord._recordType,
  ClaudeAttributionFsRecord as any,
);

File: flow_sdk/fs_records/claude/claude_settings_json/attribution.py
"""ClaudeAttributionFsRecord — attribution block from settings.json."""

from __future__ import annotations

from typing import Any, ClassVar

from flow_sdk.fs_store import Record, RecordType


class ClaudeAttributionFsRecord(Record):
    """Attribution configuration from settings.json ``attribution`` block.

    Controls how Claude Code credits itself in commits and PRs.
    """

    _record_type: ClassVar[str] = RecordType.CLAUDE_SETTINGS_JSON_ATTRIBUTION

    def __init__(self, **kwargs: Any):
        if "type" not in kwargs:
            kwargs["type"] = RecordType.CLAUDE_SETTINGS_JSON_ATTRIBUTION
        super().__init__(**kwargs)

    @classmethod
    def from_raw(cls, data: dict) -> ClaudeAttributionFsRecord:
        """Create from an ``attribution`` block."""
        rec = cls(
            commit=data.get("commit", ""),
            pr=data.get("pr", ""),
        )
        import uuid as _uuid
        rec.id = str(_uuid.uuid5(_uuid.NAMESPACE_DNS, "claude_settings_json_attribution:default"))
        return rec
